src/api/slice.cc

```diff
@@ -40,6 +40,12 @@ void slice(const FunctionCallbackInfo<Value>& args) {
     auto vSubject = args[2];
     int sliceStart = args[3]->IntegerValue(context).FromJust();
 
+    int len =  v8::Local<v8::String>::Cast(vResult)->Length();
```

Comment:
umm, slice resulting strings don't have cache issues? should the same logic be applied as in `NewTaintedString`?

Reply:
and what about `substring`, `trim`, `replace` methods?

Reply:
`substring` and `trim` methods works as expected - the result is only tainted as long as its length is greater than 0. These cases are tested in unitary tests for [trim](https://github.com/DataDog/dd-native-iast-taint-tracking-js/blob/main/test/js/trim.spec.js#L11), [substr](https://github.com/DataDog/dd-native-iast-taint-tracking-js/blob/main/test/js/substr.spec.js#L228) and [substring](https://github.com/DataDog/dd-native-iast-taint-tracking-js/blob/main/test/js/substring.spec.js#L228).

For the `replace` method, I did not found a case which results in 0 length tainted string. To get an empty string as a result, the replacer must be an empty string and must match entirely the subject string. Since empty strings cannot be tainted, the result will be untainted too.

Let me know if there is any case I'm missing for `replace`.

cc @uurien - This answer applies to your last comment too.